### DISCUSSION TITLE
Krita's GUI is broken with Kvantum since Krita 5.0.0 (a workaround is added)

### DISCUSSION THREAD

**Discussion by tsujan** [tsujan/Kvantum#746] [General]
I opened this discussion to tell Kvantum users that Krita started to have code lines like the following since its version 5.0.0:

```c++
if (style == "breeze") {
    rect.adjust(-4, -4, 0, 4);
} else if (style == "fusion") {
    rect.adjust(-2, -1, 2, 1);
}
```

Worse than that, they've assumed the spin-up and spin-down indicators are always at top right and bottom right, respectively! It means that Krita's GUI is written only for Fusion and Breeze and, as a result, parts of its GUI will be broken with Kvantum. Other Qt styles may not show the problems clearly because they use Fusion parameters for most widgets.

Nothing can be done by Kvantum to compensate for this bad practice. You could use Fusion or Breeze with Krita by unsetting the environment variable `KRITA_NO_STYLE_OVERRIDE`.

EDIT: See https://github.com/tsujan/Kvantum/discussions/746#discussioncomment-1877733

**Comment by varlesh** [comment]
This is a very bad sign and a trend. I can't imagine what the KDE desktop will look like in a couple of years with the Kvantum engine.
The new curves of the application that they create are completely unclaimed, devoid of meaning, lack of style and logic. Just compare the two players Amarok and Elisa their appearance, functionality, ease of use, etc.

**Reply by tsujan** [reply]
> This is a very bad sign

Indeed, it is. I'd say it might be a rare exception if I didn't see so many examples of bad style codes in KDE.

On the bright side, KDE only uses Qt; it can't ruin `QStyle` itself. I'm also happy that I've kept Kvantum independent of KDE.

**Comment by varlesh** [comment]
But now they are breaking old apps. Why the hell are you doing this KDE devs?

**Reply by tsujan** [reply]
> Why the hell are you doing this KDE devs?

Not all KDE devs do so. KDE is a mixture of good and bad. Unfortunately, it seems that the bad part is devouring the good one.

**Comment by tsujan** [comment]
After spending two hours in the jungle of Krita's nasty style code, I made its new "KisDoubleSliderSpinBox" as tolerable as possible:

![krita_5](https://user-images.githubusercontent.com/981076/147535075-2e056613-f9ff-48dd-814a-026c9c43d0f7.png)

I'm not sure I'll do such things again. **I'm fed up with cleaning up after KDE devs.**
